orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/pipeline/job/PreconfiguredJobStage.groovy

```diff
@@ -48,7 +48,9 @@ class PreconfiguredJobStage extends RunJobStage {
   }
 
   private Map<String, Object> overrideIfNotSetInContextAndOverrideDefault(Map<String, Object> context, PreconfiguredJobStageProperties preconfiguredJob) {
-    fields.each {
+    def fieldsToCheck = fields.collect()
```

Comment:
This is just following the pre-existing code, but I'm not a huge fan of the reflection used above...it might make more sense to a `getOverridableFields()` in the abstract class, then have the child classes override it with their additional fields and then call the super to add anything new.  That being said this is mostly pre-existing

Reply:
Agreed, I think I actually prefer that. I guess I'll take this Travis outage as an opportunity to refactor this now 😃 .